### PR TITLE
fix: origin directory prefix + added expired batch condition (MAPCO-4101)

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/cleanupCommand/cleanupManager.ts [hunk 1]
```diff
@@ -43,20 +44,22 @@ export class CleanupManager {
...
-      const tilesDirectories = this.getTilesLocation(expiredBatch);
-      await this.tileProvider.deleteDiscretes(tilesDirectories);
-      const failedDiscreteLayers = await this.mapproxy.deleteLayers(expiredBatch);
-      const completedDiscretes = expiredBatch.filter((el) => !failedDiscreteLayers.includes(el));
-      this.logger.info({
-        msg: `Complete and mark jobs as 'Completed' with file directories remove for all expired jobs`,
-        jobIds: completedDiscretes.map((job) => job.id),
-      });
-      await this.jobManager.markAsCompletedAndRemoveFiles(completedDiscretes);
+      if (expiredBatch.length > 0) {
```
Comment (asafMasa) [inline src/cleanupCommand/cleanupManager.ts]: if (expiredBatch.length == 0) { continue; }

File: src/cleanupCommand/cleanupManager.ts [hunk 2]
```diff
@@ -89,8 +91,12 @@ export class CleanupManager {
         batch: `${i + 1}/${Math.floor(notCleanedAndSuccess.length / this.discreteBatchSize + 1)}`,
         jobIds: blackListFilteredBatch.map((job) => job.id),
       });
-      const sourcesToDelete = currentBatch.filter((discrete) => !this.sourceBlackList.includes(discrete.parameters.originDirectory));
-      const ignoredSources = currentBatch.filter((discrete) => this.sourceBlackList.includes(discrete.parameters.originDirectory));
+      const sourcesToDelete = currentBatch.filter(
+        (discrete) => !this.sourceBlackList.includes(extractRootDirectory(discrete.parameters.originDirectory))
+      );
+      const ignoredSources = currentBatch.filter((discrete) =>
+        this.sourceBlackList.includes(extractRootDirectory(discrete.parameters.originDirectory))
```
Comment (asafMasa) [inline src/cleanupCommand/cleanupManager.ts]: please create a method and reuse it 
```
isIncludedInBlaclist(directory: string) {
   return this.sourceBlackList.includes(extractRootDirectory(directory));
} 
```

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/cleanupCommand/cleanupManager.ts (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/cleanupCommand/cleanupManager.ts]: fixed

File: src/cleanupCommand/cleanupManager.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/cleanupCommand/cleanupManager.ts]: fixed

**Review by asafMasa — APPROVED** [review]

**Pull request merged**